Lib/concurrent/futures/thread.py

```diff
@@ -148,4 +150,5 @@ def shutdown(self, wait=True):
         if wait:
             for t in self._threads:
                 t.join()
+        self._threads.clear()
```

Comment:
This change is not strictly needed to fix the initial bug:
https://bugs.python.org/issue31249#msg300634

But it should help to prevent leaking dangling threads if the executor object lives longer than expected.